product_name_generator/models/product_image.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import base64
+import json
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+IMAGE_TYPES = ["image/png", "image/jpeg", "image/bmp", "image/tiff"]
+
+
+class ProductImage(models.Model):
+    _inherit = "product.image"
+
+    def call_openAI(self, session):
+        response_json_str = session.call_openAI()
+        response_json = json.loads(response_json_str)
+        product_name = response_json.get("product_name")
+        self.product_tmpl_id.name = product_name
```

Comment:
An error has occurred because of the product name language.

```suggestion
       self.with_context(lang='ja_JP').product_tmpl_id.name = product_name
```
```suggestion
        self.product_tmpl_id.name = product_name
```

Reply:
The user's Odoo language is Japanese, and they use only one language in Odoo. English is disabled.

Reply:
Done